### PR TITLE
Fix getnetworkinfo XThinBlock statistics

### LINKED ISSUE #546: getnetworkinfo() RPC call does not return all thinblock info
More specifically, in rpc/net.cpp, GetThinBlockStats() the push_backs of multiple times summary let the values in the dict overwrite each other.

This is for the current dev branch. Found it while implementing a thinblock.py rpc-test.

I am going to make a PR for this issue right now.

### PR DESCRIPTION
**Pull request by awemany**
Closes #546.

### CODE REVIEW

File: qa/rpc-tests/thinblocks.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2015 The Bitcoin Core developers
+# Copyright (c) 2015-2016 The Bitcoin Unlimited developers
```
Comment (ftrader) [inline qa/rpc-tests/thinblocks.py:3]: update to 2015-2017

File: qa/pull-tester/rpc-tests.py [hunk 2]
```diff
@@ -203,6 +203,7 @@ def option_passed(option_without_dashes):
     'abandonconflict',
     'p2p-versionbits-warning',
     'importprunedfunds',
+    'thinblocks'
```
Comment (ftrader) [inline qa/pull-tester/rpc-tests.py:206]: Principal ACK on name, but I'm not really aware of thinblocks functionality being exposed apart from Xthin.
Other reviewers might correct me here.

I would have chosen xthinblocks.py but the name is fine.

File: qa/rpc-tests/thinblocks.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
```
Comment (ftrader) [inline qa/rpc-tests/thinblocks.py:9]: seems a small test so far, you might get away with just importing what you need.

There is a tool for helping figure that out - it's a bit of a PITA to use though:
https://github.com/quentinsf/dewildcard

**Review by ftrader — COMMENTED** [review]
Still need to test, but the code looks alright to me.

**Comment by sickpig** [comment]
@gandrewstone has this PR been backported to `release` ?

**Pull request merged**